src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -1450,6 +1459,17 @@ ss::future<add_paritions_tx_reply> tx_gateway_frontend::do_add_partition_to_tx(
           request, tx_errc::not_coordinator);
     }
 
+    if (stm->is_transaction_draining(request.transactional_id)) {
```

Comment:
First `add_partition_to_tx` starts a tx while the follow up `add_partition_to_tx` modifies an existing an already started tx. We should return `not_coordinator` only the first `add_partition_to_tx` but let the existing txns finish without interruption. The idea of draining is to let the inflight transactions to finish but to prevent new txes from starting.

Reply:
You need to inspect the output of `get_ongoing_tx` and reject the request only if `partitions` & `groups` are empty.